site/app/libraries/database/DatabaseQueries.php

```diff
@@ -781,6 +795,9 @@ public function createThread($markdown, $user, $title, $content, $anon, $prof_pi
 
         //Max id will be the most recent post
         $id = $this->course_db->rows()[0]["max_id"];
+        if ($announcement) {
+            $this->setAnnounced($id);
+        }
```

Comment:
Why run a separate query to update the thread as opposed to just setting the `announced` value within the `insert` above?